### ISSUE TITLE
Default value and creation of new document

### ISSUE THREAD

**Issue by PhantomYdn** [orientechnologies/orientdb#4535]
Guys,

From common logic perspective `new ODocument(classA)` should already have all fields with default values prepopulated, but OrientDB requires storing of ODocument for population of default values. And more over: this done in strange for this taks method ODocument.validateField().

I suggest:
1) Refactor ODocument.validateField() (might be in parallel with #3429)
2) Allow to prepopulate ODocument by default values on creation, but not onSave

So, what do you think?

**Comment by luigidellaquila** [comment]
IMHO pre-populating ODocument with default values is a good solution. I just have to check how it will affect backward compatibility.

**Comment by lvca** [comment]
:+1

**Comment by PhantomYdn** [comment]
Guys, please consider PR. All tests are OK.

**Comment by lvca** [comment]
The call to the populateDefaultValues() should be done also on setClass() and setClassname() right?

**Comment by PhantomYdn** [comment]
My idea was that once document created we can do with the object what ever we want, but fields values remains the same. Other wise there are lots of additional questions like: what to do with the a prepopulated field if super class was changed? Clean it or keep it? What to do if new class has the same property by with some default value? And so on.

**Comment by lvca** [comment]
The problem is that new ODocument(<class>) is semantically equivalent to: new ODocument(<class>)

**Comment by PhantomYdn** [comment]
My idea was that in case 

```java
new ODocument().setClassname(<class>);
```

document will not have default values, because class was defined after creation of a document, so population of default values should be explicitly invoked.

In case 

```java
new ODocument("ClassA").setClassname("ClassB");
```
document will have default values from ClassA and changing to ClassB will preserve already set values.

If you disagree: just let me know proper way to resolve that.

**Comment by PhantomYdn** [comment]
Seems to be fixed in #4552. Closing?
